### PR TITLE
Fix for #357 KV clusters do not work

### PR DESCRIPTION
**Pull request by danielin917**
monmap naming scheme extended across all calls to get_mon_config

Signed-off-by: Daniel Lin <danielin@umich.edu>

### CODE REVIEW

**Comment by immesys** [comment]
Do you not need to modify [L65](https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/config.kv.sh#L65) , [L67](https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/config.kv.sh#L67), [L77](https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/config.kv.sh#L71) [L73](https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/config.kv.sh#L73) [L108](https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/config.kv.sh#L108) and [L117](https://github.com/ceph/ceph-docker/blob/master/ceph-releases/jewel/ubuntu/14.04/daemon/config.kv.sh#L117) as well?

**Comment by danielin917** [comment]
You're correct. Also, I'm not very familiar with running the containerized clusters with kv stores but I assumed the kviator get and put paths would have to change also to be cluster based. If someone could confirm that or let me know of any other fixes that would be great

**Comment by immesys** [comment]
Actually the key (for kviator) does contain the cluster name, so those are okay.,

File: ceph-releases/hammer/ubuntu/14.04/daemon/config.kv.sh [hunk 1]
```diff
@@ -51,13 +51,13 @@ function get_mon_config {
     kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/bootstrapRgwKeyring > /var/lib/ceph/bootstrap-rgw/${CLUSTER}.keyring
 
 
-    if [ ! -f /etc/ceph/monmap ]; then
+    if [ ! -f /etc/ceph/monmap-${CLUSTER} ]; then
       echo "Monmap is missing. Adding initial monmap..."
-      kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/monmap > /etc/ceph/monmap
+      kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/monmap-${CLUSTER} > /etc/ceph/monmap-${CLUSTER}
```
Comment (immesys) [inline ceph-releases/hammer/ubuntu/14.04/daemon/config.kv.sh]: actually I don't think you need to modify the key here. CLUSTER is part of CLUSTER_PATH

File: ceph-releases/hammer/ubuntu/14.04/daemon/config.kv.sh [hunk 2]
```diff
@@ -102,7 +102,7 @@ function get_mon_config {
     kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} put ${CLUSTER_PATH}/bootstrapMdsKeyring - < /var/lib/ceph/bootstrap-mds/${CLUSTER}.keyring
     kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} put ${CLUSTER_PATH}/bootstrapRgwKeyring - < /var/lib/ceph/bootstrap-rgw/${CLUSTER}.keyring
 
-    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} put ${CLUSTER_PATH}/monmap - < /etc/ceph/monmap
+    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} put ${CLUSTER_PATH}/monmap-${CLUSTER} - < /etc/ceph/monmap-${CLUSTER}
```
Comment (immesys) [inline ceph-releases/hammer/ubuntu/14.04/daemon/config.kv.sh]: same here

**Comment by danielin917** [comment]
Makes sense, I've reverted that change for Jewel, Hammer and Infernalis

**Comment by leseb** [comment]
I think we are good with this one, @immesys any other comments from you?

**Comment by immesys** [comment]
Looks good, reading it over. @leseb What are the contribution guidelines with regards to the targeted release? I've only been working with jewel, do you want PRs to backport changes to hammer and infernalis too? Or do you want features to only go on jewel?

**Comment by leseb** [comment]
@danielin917 at this point, I'm only pushing new features in Jewel and bug fixes in other ceph releases. So we are keeping our focus on Jewel.

**Pull request merged**